### ISSUE TITLE
_p9k_human_readable_bytes()'s implementation giving undesirable results

### ISSUE THREAD

**Issue by jsjohnst** [romkatv/powerlevel10k#1664]
On machines with >100GB RAM, the implementation of `_p9k_human_readable_bytes` gives a poor experience. As an example, on a machine with 128GB RAM, it shows free memory as `0.11T` rather than a much more friendly `111.33G`. The reason appears to be that the function is enforcing any number greater than 100 to become the next unit suffix larger producing a value less than 1 of the unit. This value also isn't configurable forcing one to edit "internal" implementation details.

If this isn't going to be made customizable, I would ask that something like the following be done instead:

```
function _p9k_human_readable_bytes() {
  typeset -F 2 n=$1
  local suf
  for suf in $__p9k_byte_suffix; do
    (( n < 512 )) && break
    (( n /= 1024 ))
  done
  _p9k__ret=${${n%%0#}%.}$suf
}
```

basically saying you need to have greater than 0.5 of the next larger unit before it forces the upgrade to the next unit size larger.

**Comment by romkatv** [comment]
Thanks for the bug report. Fixed.
